pkg/zkproofs/ciphertext_validity.go

```diff
@@ -0,0 +1,163 @@
+package zkproofs
+
+import (
+	crand "crypto/rand"
+	"encoding/json"
+	"errors"
+	"github.com/coinbase/kryptology/pkg/core/curves"
+	"github.com/sei-protocol/sei-cryptography/pkg/encryption/elgamal"
+	"math/big"
+)
+
+// CiphertextValidityProof represents a zero-knowledge proof that a ciphertext is valid.
+type CiphertextValidityProof struct {
+	Commitment1 curves.Point
+	Commitment2 curves.Point
+	Challenge   curves.Scalar
```

Comment:
I think Challenge here shouldn't be passed in the Proof, it should be independently generated by the Validator for safety

Reply:
ok, good catch, I think this was pasted from PoC

Reply:
Fixed